tests/unit/test_dns_server.py

```diff
@@ -460,3 +466,19 @@ def test_no_resolv_conf_overwriting_on_host(self, tmp_path: Path, monkeypatch):
 
         assert "nameserver 127.0.0.1" not in new_contents.splitlines()
         assert "nameserver 127.0.0.11" in new_contents.splitlines()
+
+    def test_host_prefix_no_subdomain(
```

Comment:
Thank you for adding this test step, this will be great for preventing regressions and making sure we handle new services!